esp/src/src-react/components/Scopes.tsx

```diff
@@ -71,39 +87,19 @@ export const Scopes: React.FunctionComponent<ScopesProps> = ({
     const [, { currencyCode }] = useBuildInfo();
 
     React.useEffect(() => {
-        const files = [];
         if (scope === ".") {
-            dfuService.DFUFileView({ Scope: "" }).then(({ DFULogicalFiles }) => {
-                const _scopes = DFULogicalFiles?.DFULogicalFile.filter(file => file.isDirectory && file.Directory !== ".") || [];
-
-                dfuService.DFUFileView({ Scope: "." }).then(({ DFULogicalFiles }) => {
-                    _scopes.forEach((scope, idx) => {
-                        scope["__hpcc_id"] = scope.Directory + "_" + idx;
-                        files.push(scope);
-                    });
-                    DFULogicalFiles?.DFULogicalFile.forEach((file, idx) => {
-                        file["__hpcc_id"] = file.Name + "_" + idx;
-                        file["Name"] = file.Name.split("::").pop();
-                        files.push(file);
-                    });
-                    setData(files);
-                });
+            dfuService.DFUFileView({ Scope: "" }).then(async ({ DFULogicalFiles }) => {
+                const scopes = DFULogicalFiles?.DFULogicalFile.filter(file => file.isDirectory && file.Directory !== ".") || [];
+                const rootFiles = await dfuService.DFUFileView({ Scope: "." });
+                const files = rootFiles?.DFULogicalFiles?.DFULogicalFile || [];
+                setData(mergeFileData(scopes, files));
             });
         } else {
             setScopePath(scope.split("::"));
             dfuService.DFUFileView({ Scope: scope }).then(({ DFULogicalFiles }) => {
-                const _scopes = DFULogicalFiles?.DFULogicalFile.filter(file => file.isDirectory && file.Directory !== ".") || [];
-                const _files = DFULogicalFiles?.DFULogicalFile.filter(file => !file.isDirectory) || [];
-                _scopes.forEach((scope, idx) => {
-                    scope["__hpcc_id"] = scope.Directory + "_" + idx;
-                    files.push(scope);
-                });
-                _files.forEach((file, idx) => {
-                    file["__hpcc_id"] = file.Name + "_" + idx;
-                    file["Name"] = file.Name.split("::").pop();
-                    files.push(file);
-                });
-                setData(files);
+                const scopes = DFULogicalFiles?.DFULogicalFile.filter(file => file.isDirectory && file.Directory !== ".") || [];
```

Comment:
`|| []`` is not needed (filter always returns an array)
You also use the same filter condition here and below, could be commoned up